envoy/router/router.h

```diff
@@ -610,6 +610,7 @@ class VirtualCluster {
 
 class RateLimitPolicy;
 class Config;
+class SharedConfig;
```

Comment:
comment: split the interfaces that needn't depend on router matcher out to the `SharedConfig`. Then every virtualhost could keep a reference to the `SharedConfig`.
When the entire route config is destroyed, the part of `SharedConfig` will still alive until all active requests are completed.